### PR TITLE
Add thread message positions

### PR DESCRIPTION
**Pull request by freya022**
[contributing]: https://jda.wiki/contributing/contributing/

## Pull Request Etiquette

- [X] I have checked the PRs for upcoming features/bug fixes.
- [X] I have read the [contributing guidelines][contributing].

### Changes

- [X] Internal code
- [X] Library interface (affecting end-user code) 
- [X] Documentation
- [ ] Other: \_____ 

Closes Issue: #2305 

## Description

This adds the `position` property from the message object.

### CODE REVIEW

**Comment by freya022** [comment]
Older messages have missing positions, I assume this is before discord added the property, should it be documented ?

**Review by DV8FromTheWorld — APPROVED** [review]
LGTM

**Review by Mysterious-Dev — APPROVED** [review]

File: src/main/java/net/dv8tion/jda/api/entities/Message.java [hunk 1]
```diff
@@ -360,6 +360,25 @@ default Message getReferencedMessage()
...
     Member getMember();
 
+    /**
+     * Returns the approximate position of this message in a {@link ThreadChannel}.
+     * <br>This can be used to estimate the relative position of a message in a thread, by comparing against {@link ThreadChannel#getTotalMessageCount()}.
+     *
+     * <p><b>Notes:</b>
+     * <ul>
+     *     <li>The position might contain gaps or duplicates.</li>
+     *     <li>The position is not set on messages sent earlier than July 19th 2022, and will return -1.</li>
```
Comment (Chew) [inline src/main/java/net/dv8tion/jda/api/entities/Message.java:370]:
```suggestion
     *     <li>The position is not set on messages sent earlier than July 19th, 2022, and will return -1.</li>
```

File: src/main/java/net/dv8tion/jda/internal/entities/ReceivedMessage.java [hunk 2]
```diff
@@ -332,6 +334,15 @@ public Member getMember()
         return member;
     }
 
+    @Override
+    public int getApproximatePosition()
+    {
+        if (!getChannelType().isThread())
+            throw new IllegalStateException("This message was not sent in a thread.");
```
Comment (rittvic) [inline src/main/java/net/dv8tion/jda/internal/entities/ReceivedMessage.java:341]:
```suggestion
        if (!getChannelType().isThread()) 
        {
            throw new IllegalStateException("This message was not sent in a thread.");
        }
```
I would place curly brackets here to remain consistent with the JDA [code style](https://github.com/DV8FromTheWorld/JDA/wiki/6%29-JDA-Structure-Guide#indentation-and-brackets)

Comment (freya022) [reply src/main/java/net/dv8tion/jda/internal/entities/ReceivedMessage.java:341]: Maintainers does not want pointless spaces, such as these one liners

**Review by Taz03 — APPROVED** [review]

**Pull request merged**